src/scripts/app.js

```diff
@@ -37,12 +49,13 @@ class App extends JSXComponent {
         return response.json();
       })
       .then((response) => {
-        if (response.statusCode >= 400) {
+        
+        // Response doesn't contain status code in here.
+        if (!response.hasOwnProperty("html") ) {
```

Comment:
Also @zenorocha another interesting bit is that `statusCode` will tell you that you had a sucessful HTTP request and not necessarily that your API is the one answering. A captive portal, intranet paranoid firewall, stupid proxy, all of them might try to hijack and redirect this connection which would lead to status codes below 400 (200 or 3xx) and break the extension. **I know this is such an edge case that might never happen**, still if we check for the presence of a known property from the API, we eliminate all potential edge cases and only treat successful results from the API as such.

Am I too paranoid? 🐹